app/controllers/rentals_controller.rb

```diff
@@ -0,0 +1,75 @@
+class RentalsController < ApplicationController
+
+  def create
+    movie = Movie.find_by_title(params[:title])
+
+    if movie.nil? || movie.available_inventory < 1
+      id = 0
+    else
+      id = movie.id
+    end
+
+    @rental = Rental.new(movie_id: id, customer_id: rental_params[:customer_id], due_date: rental_params[:due_date] )
+
+    if @rental.save
+      # movie.inventory -= 1
+      # movie.save
+      render status: :ok, json: @rental.as_json
```

Comment:
You might want to filter what's returned here to eliminate the created_at and updated_at timestamps from the database record. They seem to be in a different timezone than the local user's timezone (i.e., for me they said it was already tomorrow, "2017-06-05T03:08:04.102Z", while the checkout date was today as expected, "2017-06-04T00:00:00.000Z"). It's not a big deal but could be somewhat confusing from the API user's standpoint.